### PR TITLE
Add feature: sort tasks by deadline

### LINKED ISSUE #79: Add feature: sort by deadline

### PR DESCRIPTION
**Pull request by QY-H00**
This feature works if deadline is not optional.

### CODE REVIEW

File: src/main/java/seedu/module/model/task/UniqueTaskList.java [hunk 1]
```diff
@@ -70,6 +70,13 @@ public void setTask(Task target, Task editedTask) {
         internalList.set(index, editedTask);
     }
 
+    /**
+     * Replaces the tasks in the list by deadline.
```
Comment (yhtMinceraft1010X) [inline src/main/java/seedu/module/model/task/UniqueTaskList.java]: I wouldn't describe sorting as replace task in javadoc.

**Review by yhtMinceraft1010X — COMMENTED** [review]
LGTM overall, although I beg to differ with regards to the javadoc comment for sortTasks() under UniqueTaskList.java

File: src/main/java/seedu/module/model/task/UniqueTaskList.java (hunk 1 above)
Comment (QY-H00) [reply src/main/java/seedu/module/model/task/UniqueTaskList.java]: Fixed.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -170,6 +172,12 @@ Examples:
 *  `edit 1 d/Eat Biscuits` Edits the description of the 1st task to `Eat Biscuits`.
 *  `edit 2 d/Eat Biscuits b/2021-03-21T10:10:10` Edits the description of the 2nd task to be `Eat Biscuits` and its deadline to date `2021-03-21` and time `10:10:10`.
 
+### Sorting tasks : `sort`
+
+Sorts the list of all tasks added by deadline.
```
Comment (umergta) [inline docs/UserGuide.md:177]: Maybe can paraphrase to "sorts the list of tasks according to the deadline of the task. Tasks with deadlines approaching soon are displayed at the top while tasks with no deadlines are displayed at the bottom"

**Review by umergta — COMMENTED** [review]
overall LGTM! just gotta work on the task comparator for tasks with no deadlines (as discussed)
Solid work qiyuan :)

**Review by AiwassPrime — COMMENTED** [review]
Tests LGTM

**Review by figo2127 — COMMENTED** [review]
LGTM!

**Pull request merged**